### DISCUSSION TITLE
[Feature Request]: Anticipate webui compatibility issues with FLAC files.

### DISCUSSION THREAD

**Discussion by korzhyc** [navidrome/navidrome#3028] [Ideas]
I've noticed that in my collection (not a small one) some files are not playing through the browser/webui.

I found out that the cause is an embedded cover with an empty value for the mime_type parameter.
I also managed to find out that only flac files are affected by the problem. For mp3/opus/ogg the parameter is not that important.

The same problem is described here #2437.
I thought, since ND is digging through the tags, maybe it will hint to us that something might be wrong with the file.

And that's my request:
Add an option to check mime_type of embedded covers for flac files and log file names in case of abnormal values (empty or not png/jpg). (I assume that this should be done during scan process if `embedded` is set in `CoverArtPriority`).

**Comment by korzhyc** [comment]
Upd:
It looks like the cover_type value can also cause problems.
[Here](https://github.com/navidrome/navidrome/issues/2960#issuecomment-2078875788) is an example of the file where cover_data and cover_mime are present, but cover_type is empty.

**Comment by Szeraax** [comment]
Is your request that errors with playback get shown in the web UI instead of in the log? Or that ND will auto-skip items that won't play on that client? Or just give a little warning/toast anytime files aren't following a spec? Multiple of those? Other?

**Reply by korzhyc** [reply]
I'd like to see a complete list of defective files in the ND log, right after the scan.
